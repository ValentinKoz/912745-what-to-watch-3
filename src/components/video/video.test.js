import React from "react";
import renderer from "react-test-renderer";
import Video from "./video.jsx";

const film = {
  genre: ``,
  title: `Parasite`,
  releaseDate: 0,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  bgPoster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  ratingScore: ``,
  ratingLevel: ``,
  ratingCount: 1000,
  text: ``,
  director: ``,
  starring: ``,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
};

it(`Render Video`, () => {
  const tree = renderer
    .create(<Video
      poster={film.bgPoster}
      videoSrc={film.video}
      isPlaying={false}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
      .toJSON();

  expect(tree).toMatchSnapshot();
});
