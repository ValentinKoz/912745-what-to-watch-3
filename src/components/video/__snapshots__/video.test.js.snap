// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Video 1`] = `
<div
  className="small-movie-card__image"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <video
    height="175"
    muted={true}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    width="280"
  >
    <source
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
    />
  </video>
</div>
`;
